Update model definitions

tomwallis opened the pull request:
Updates demo models to use the new required behaviour of setting obs=None when generating predictions. Also adds demos of prior and posterior predictive distributions.

Helps to address [this](https://github.com/ag-perception-wallis-lab/numpyro-oop/issues/2).

tomwallis committed:
Update model definitions

Updates demo models to use the new required behaviour
of setting obs=None when generating predictions. Also
adds demos of prior and posterior predictive distributions.

pixelsandpointers committed:
Update and rename demo_1.ipynb to demo.ipynb

pixelsandpointers committed:
renamed folder scripts -> examples

pixelsandpointers commented:
@tomwallis so I renamed a few things and added a link to open the notebook in Colab to play around with zero local setup required.

Check if this works out for you.

tomwallis commented:
> @tomwallis so I renamed a few things and added a link to open the notebook in Colab to play around with zero local setup required.
> 
> Check if this works out for you.

Thanks, this works! Let's gooo!
